### ISSUE TITLE
Version 1.1.0 .pivotal throws Node exception when target SQL Server is unreachable

### ISSUE THREAD

**Issue by cpage-pivotal** [Azure/meta-azure-service-broker#70]
Running with Ops Manager 1.18.10 on Azure. Installing the tile fails with the following messages:

```
cf logs azure-service-broker-1.1.0 --recent
Connected, dumping recent logs for app azure-service-broker-1.1.0 in org system / space azure-service-broker-space as system_services...

2016-12-08T02:34:55.08+0000 [APP/0]      OUT > meta-azure-service-broker@1.0.0 start /home/vcap/app
2016-12-08T02:34:55.67+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mStarting to collect the service offering and plans of each service module...
2016-12-08T02:34:55.67+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39m[34m[1msqlserver[22m[39m[32m - [39mLoading schema
2016-12-08T02:34:55.75+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mAdding listeners for the service azure-documentdb...
2016-12-08T02:34:55.76+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mAdding listeners for the service azure-rediscache...
2016-12-08T02:34:55.77+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mAdding listeners for the service azure-servicebus...
2016-12-08T02:34:55.77+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mAdding listeners for the service azure-sqldb...
2016-12-08T02:34:55.79+0000 [APP/0]      OUT [90m2:34:55 AM[39m[32m - [39mINFO [32m - [39m[34m[1mMeta Azure Service Broker[22m[39m[32m - [39mAdding listeners for the service azure-storage...
2016-12-08T02:34:55.90+0000 [APP/0]      ERR timers.js:327
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     throw new TypeError('"callback" argument must be a function');
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     ^
2016-12-08T02:34:55.90+0000 [APP/0]      ERR TypeError: "callback" argument must be a function
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at exports.setTimeout (timers.js:327:11)
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at /home/vcap/app/lib/broker/db/sqlserver/sqlserver.js:19:16
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at /home/vcap/app/node_modules/mssql/lib/main.js:287:20
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at /home/vcap/app/node_modules/mssql/lib/tedious.js:432:20
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at /home/vcap/app/node_modules/generic-pool/lib/generic-pool.js:340:9
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at Connection.<anonymous> (/home/vcap/app/node_modules/mssql/lib/tedious.js:381:26)
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at Connection.g (events.js:292:16)
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at emitOne (events.js:96:13)
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at Connection.emit (events.js:188:7)
2016-12-08T02:34:55.90+0000 [APP/0]      ERR     at Connection.socketError (/home/vcap/app/node_modules/tedious/lib/connection.js:571:14)
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Linux 4.4.0-45-generic
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! argv "/home/vcap/app/.heroku/node/bin/node" "/home/vcap/app/.heroku/node/bin/npm" "start"
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! node v7.2.0
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! npm  v3.10.9
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! code ELIFECYCLE
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! meta-azure-service-broker@1.0.0 start: `node index.js`
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Exit status 1
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! 
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Failed at the meta-azure-service-broker@1.0.0 start script 'node index.js'.
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Make sure you have the latest version of node.js and npm installed.
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! If you do, this is most likely a problem with the meta-azure-service-broker package,
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! not with npm itself.
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Tell the author that this fails on your system:
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR!     node index.js
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! You can get information on how to open an issue for this project with:
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR!     npm bugs meta-azure-service-broker
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! Or if that isn't available, you can get their info via:
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR!     npm owner ls meta-azure-service-broker
2016-12-08T02:34:55.97+0000 [APP/0]      ERR npm ERR! There is likely additional logging output above.
2016-12-08T02:34:55.98+0000 [APP/0]      ERR npm ERR! Please include the following file with any support request:
2016-12-08T02:34:55.98+0000 [APP/0]      ERR npm ERR!     /home/vcap/app/npm-debug.log
2016-12-08T02:34:55.98+0000 [APP/0]      OUT Exit status 1

```

**Comment by bingosummer** [comment]
@cpage-pivotal This is a bug of the broker. It will be triggered when the sql server can't be connected. We will fix it ASAP.

Meanwhile, could you please confirm the following items. If the sql server can be connected successfully, the issue should be gone.

1. Check the connection information.

  ```
  AZURE_BROKER_DATABASE_SERVER: REPLACE-ME
  AZURE_BROKER_DATABASE_USER: REPLACE-ME
  AZURE_BROKER_DATABASE_PASSWORD: REPLACE-ME
  AZURE_BROKER_DATABASE_NAME: REPLACE-ME
  ```

  You can check it via `mssql --server "<server-name>.database.windows.net" --database <database> --user <username>@<server-name> --pass <pass> --encrypt`.

2. Check the firewall rules.

  You need to make sure the sql server can be connected. FYI: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-configure-firewall-settings

**Comment by cpage-pivotal** [comment]
Thanks, I will verify connectivity!

**Comment by bingosummer** [comment]
@cpage-pivotal 
How is it going?
Here is a pre-release v1.2.0.rc1. In this version, if it fails to connect the database, it will retry and give a readable error message. Would you like to try it? 
https://github.com/Azure/meta-azure-service-broker/releases/download/v1.2.0.rc1/azure-service-broker-1.2.0.rc1.pivotal

**Comment by cpage-pivotal** [comment]
Thanks, I cleared up the connectivity issues, and everything worked great. I have changed the issue title to properly reflect the problem. I will test the 1.2.0 tile when I get a chance.

**Comment by bingosummer** [comment]
Closing it as it's fixed in `v1.2.0`.
Feel free to reopen it if you still hit the issue.
